libs/common-services/src/main/java/uk/gov/di/ipv/core/library/tracing/TracingHttpClient.java

```diff
@@ -100,6 +103,25 @@ public <T> HttpResponse<T> send(
         }
     }
 
+    @Override
+    public <T> HttpResponse<T> send(
+            HttpRequest request, HttpResponse.BodyHandler<T> responseBodyHandler)
+            throws IOException, InterruptedException {
+        try {
+            return sendWithTracing(request, responseBodyHandler);
+        } catch (IOException e) {
+            // We see occasional HTTP/2 GOAWAY messages from AWS when connections last
+            // for a long time (>1 hour). Retry them once, to recreate the connection.
+            if (e.getMessage().contains("GOAWAY received")) {
```

Comment:
This is a bit unsatisfactory, but I couldn't see a cleaner solution... The messages look like `/169.254.76.1:51122: GOAWAY received` (where the port varies per request)